experimental/AlgebraicStatistics/src/PhylogeneticModels-functions.jl

```diff
@@ -0,0 +1,307 @@
+###################################################################################
+#
+#       Auxiliary graph functions
+#
+###################################################################################
+
+interior_nodes(graph::Graph) = findall(>=(1), outdegree(graph))
+
+leaves(graph::Graph) = findall(iszero, outdegree(graph))
+n_leaves(graph::Graph) = length(leaves(graph))
+n_leaves(pt::PhylogeneticTree) = n_leaves(adjacency_tree(pt))
+root(pt::PhylogeneticTree) = _root(adjacency_tree(pt))
+
+
+function roots(graph::Graph)
+  n_parents = [length(inneighbors(graph, v)) for v in 1:n_vertices(graph)]
+  return findall(iszero, n_parents)
+end
+
+root(graph::Graph) = roots(graph)[1]
```

Comment:
```suggestion
root(graph::Graph) = only(roots(graph))
```
This would make it error if there are multiple roots.